### DISCUSSION TITLE
access control bug

### DISCUSSION THREAD

**Discussion by softwaredevelop** [community/community#29415] [Codespaces] [answered]
1.

```bash
sudo chown codespace:root /var/run/docker.sock && ls -la /var/run/docker.sock
srw-rw---- 1 codespace root 0 Aug 14 11:46 /var/run/docker.sock
```

after reconnect to codespace:
```bash
ls -la /var/run/docker.sock
srw-rw---- 1 root 800 0 Aug 14 12:01 /var/run/docker.sock
```

2.

```bash
sudo setfacl -m user:codespace:rw /var/run/docker.sock
getfacl /var/run/docker.sock
getfacl: Removing leading '/' from absolute path names
# file: var/run/docker.sock
# owner: root
# group: 800
user::rw-
user:codespace:rw-
group::rw-
mask::rw-
other::---
```

after reconnect to codespace:
```bash
getfacl /var/run/docker.sock
getfacl: Removing leading '/' from absolute path names
# file: var/run/docker.sock
# owner: root
# group: 800
user::rw-
group::rw-
other::---
```
😭

**Comment by thomas-sickert** [comment, marked as answer]
Hey 👋 So, this doesn't seem like an access control bug. 

Codespaces uses docker to manage/run your devcontainer. I would expect the docker socket to be modified/provisioned by codespaces' infrastructure, especially when you're disconnecting from/stopping a codespace.

I'd say to proceed with caution, but If you'd like these changes to "persist" you could put the steps in your `.bashrc` (or the equivalent for the shell you use)

**Reply by softwaredevelop** [reply]
I modified $HOME/.bashrc in codespace and user access is working (Docker in Docker)

```bash
USERNAME="codespace"
TARGET_SOCKET="/var/run/docker.sock"

tee --append /home/${USERNAME}/.bashrc \
  <<EOF

function sudoIf() {
  if [ "\$(id -u)" -ne 0 ]; then
    sudo "\$@"
  else
    "\$@"
  fi
}
if [ -e "${TARGET_SOCKET}" ]; then
  sudoIf setfacl --modify=user:${USERNAME}:rw ${TARGET_SOCKET}
fi
EOF

```
Without this the codspace after disconnect/stop/reconnect into a "default state". Because Codespace BETA?
